### DISCUSSION TITLE
GridifyQueryBuilder addCondition() with CustomOperator in TS client

### DISCUSSION THREAD

**Discussion by kamsiuman** [alirezanet/Gridify#225] [Q&A] [answered]
Hello All,

I have developed a dot-net backend with the customer operator InOperator with the guidance .   [here](https://alirezanet.github.io/Gridify/guide/filtering#custom-operators)

Now I am trying to using the GridifyQueryBuilder() in TS;  it seems to be in the addCondition() , the operator only accepts the enum type in ConditionalOperator [here](https://alirezanet.github.io/Gridify/guide/extensions/gridify-client#conditional-operators)

```
import { GridifyQueryBuilder, ConditionalOperator as op } from 'gridify-client';

.....

    const query = new GridifyQueryBuilder()
      .setPage(2)
      .setPageSize(10)
      .addCondition('name', op.Contains, 'joe')
      .addCondition('countries', <????>, '1;3;4')
      .build();
```

one workaround I could think of is to do manual-fix the query.filter after the build() and I wonder if there is better way I might have missed. 

Any suggestions/comments are welcome! 

Thanks,
--alex

**Comment by alirezanet** [comment]
Hi @kamsiuman,
Please try using gridify-client version `2.0.2`; it should resolve your issue. In this version, you have the ability to pass your custom operator as string to the `addCondition` method.

**Reply by alirezanet** [reply]
More info:
https://github.com/alirezanet/Gridify/releases/tag/c-v2.0.2

**Comment by kamsiuman** [comment, marked as answer]
hi @alirezanet,

It works perfectly ! thank you very much!

Thank you for the **quick** response as well ! 👍 

Thanks,
--alex
